### PR TITLE
Week 15 - project Todos Sofia 

### PR DESCRIPTION
**Pull request by Sofarz**
Link to my deployed project: https://quirky-colden-bd6702.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/AddTaskForm.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
```
Comment (magicvilla) [inline code/src/components/AddTaskForm.js:1]: Very clean and readable code Sofia 😄  There are some places where you add semi-colon and places where you don't. Remember to be consistent with the semi-colons 💯

File: code/src/components/ClearedPage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                alt='icecream' 
+            /> 
+            <ExtraTextClearPage>
+                Or maybe you JUST thought of something more you need to do?
+            </ExtraTextClearPage>
+        </PlaceholderPage>
+    )
+}
+
+export default ClearedPage
```
Comment (magicvilla) [inline code/src/components/ClearedPage.js:26]: I Love your Cleared Page! Its really neat looking! Your components continue to be clean and clear. Your code is really readable!  👍 😃

File: code/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                <CompletedTasksText>
+                    Completed: {completedTasks}/{totalNumberOfTasks}
+                </CompletedTasksText>
+                <AllDoneButton onClick={onClickClearAll}>
+                    All done!
+                </AllDoneButton>
+            </HeaderCounterButtonContainer>
+        </HeaderContainer>
+    )
+}
```
Comment (magicvilla) [inline code/src/components/Header.js:39]: I like how you have structured your styled components in separate files and then imported them. Next time i will copy you and try to do the same!

File: code/src/components/Task.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { useDispatch } from 'react-redux'
+import Checkbox from '@material-ui/core/Checkbox'
+import IconButton from '@material-ui/core/IconButton';
+import DeleteIcon from '@material-ui/icons/Delete';
+
```
Comment (magicvilla) [inline code/src/components/Task.js:6]: Same here 😸  Try to be consistent with the semi-colons

File: code/src/components/Task.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                />
+                <p className={completed === true ? 'task-complete' : 'task-not-complete'}>
+                    {task.content}
+                </p>
+            </TaskCheckboxContainer>
+            <IconButton onClick={() => dispatch(tasks.actions.removeTask(task.id))} aria-label="delete">
+                <DeleteIcon />
+            </IconButton>
+        </TodoTask>
+    )
```
Comment (magicvilla) [inline code/src/components/Task.js:30]: Nice icons with material UI! And thank you for introducing them to me as well 💯  Love the design and use of them!

File: code/src/components/TaskList.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    return (
+        <List>
+            {items.map((task) => (
+                <Task key={task.id} task={task} completed={task.completed} />
+            ))}
+        </List>
+    )
+}
+
+export default TaskList
```
Comment (magicvilla) [inline code/src/components/TaskList.js:19]: Your tasklist is so clean! And i really like the idea that you have a separate component for the task list and for the task item it self! I will maybe steal this idea for my own project 😉

File: code/src/components/TodoApp.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            <FormListContainer>
+                <AddTaskForm />
+                {totalNumberOfTasks === 0 ? <ClearedPage /> : <TaskList />}
+            </FormListContainer>
+            <TotalTaskContainer>
+                {totalNumberOfTasks === 0 ? !<TotalTasks /> : <TotalTasks />}
+            </TotalTaskContainer>
+        </MainAppContainer>
+    )
+}
```
Comment (magicvilla) [inline code/src/components/TodoApp.js:30]: Once again, so clean and so readable! Great job sofia 👍 💯 😄

File: code/src/components/TotalTasks.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+    return (
+            <DeleteAllButton 
+                onClick={onClickDeleteAll}
+            >   
+                Delete all!
+            </DeleteAllButton>
+    )
+}
+
```
Comment (magicvilla) [inline code/src/components/TotalTasks.js:22]: Nice done with the delete all button and your reducer! I would like to implement this as well!

File: code/src/index.css [hunk 9]
```diff
@@ -5,9 +5,37 @@ body {
...
+
+@keyframes rotation {
+  from {
+    transform: rotate(-25deg);
+  }
+  to {
+    transform: rotate(25deg);
+  }
+}
+
```
Comment (magicvilla) [inline code/src/index.css:33]: Love the animation of the ice-cream 😄  Is it to early for ice-cream? Makes me in the mood for it 😉

File: code/src/reducers/tasks.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                return {
+                    ...task, 
+                    completed: true
+                }
+            })
+            store.items = clearAll
+        }
+    }
+})
+
```
Comment (magicvilla) [inline code/src/reducers/tasks.js:50]: Clean and nice with comments to explain more what the code line does 💯

**Review by magicvilla — APPROVED** [review]
Sofia! GREAT JOB with your to-do app! I love your design and your styling is very organized. I especially like your structuring with the styled components in separate folders and then imported as components. Im going to copy this for the future! 

Your code is super clear and readable both for newbies and for oldies 😉 . 
You are good at name giving for functions, variables and class names. This is something i need to work on!

My feedback to you is to be consistent with your code based on semi-colons. Stick to one of them (either with or without). 

Great Job! Cant wait to see your other projects 🥇

File: code/src/components/AddTaskForm.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+				fullWidth
+				value={value}
+				onChange={(event) => setValue(event.target.value)}
+			/>
+			<SubmitButton type='submit'>
+				<span role="img" aria-label="add button">
+					➕
+				</span>
+			</SubmitButton>
+		</AddTodoForm>
```
Comment (KarolCislowski) [inline code/src/components/AddTaskForm.js:40]: Please set your indentation to 2 spaces in your Prettier's settings.

File: code/src/components/Task.js (hunk 4 above)
Comment (KarolCislowski) [inline code/src/components/Task.js:6]: This is another thing that you should set in the Prettier configuration.

**Review by KarolCislowski — APPROVED** [review]
Nice app with clean interface. Please check your prettier and set it properly.

**Pull request closed without merging**